### PR TITLE
Added GPU versions of volume and face reference kernels.

### PR DESCRIPTION
**Pull request by sriharshakandala**

### CODE REVIEW

**Comment by sriharshakandala** [comment]
bors try

**Comment by sriharshakandala** [comment]
bors try

**Comment by sriharshakandala** [comment]
bors try

**Comment by sriharshakandala** [comment]
bors try-

**Comment by sriharshakandala** [comment]
bors try

File: .gitignore [hunk 1]
```diff
@@ -1,5 +1,9 @@
 Manifest.toml
 
+# Temporary
+*.DS_Store
+*.swp
```
Comment (jakebolewski) [inline .gitignore:5]: you should add these to your personal / user `.gitignore` not the project's

**Comment by sriharshakandala** [comment]
bors try

**Comment by sriharshakandala** [comment]
bors try

**Comment by sriharshakandala** [comment]
bors try-

**Comment by sriharshakandala** [comment]
bors try

**Pull request merged**